documentation/release_notes.markdown

```diff
@@ -8,42 +8,72 @@ canonical: "/puppetdb/latest/release_notes.html"
-  you have to wait while these run after upgrading. Before upgrading
-  you should also ensure you have as much free disk space as you have
-  data, as migrations often require temporary duplication of table
-  data to succeed.
-
-* You may notice some additional system load for the first 30 to 60
-  minutes once the application has started upgrading. This is
-  expected, and is due to internal changes in the way we check if
-  catalogs are up-to-date.
+### Upgrading
+
+* This version of PuppetDB adopts the AIO layout to be consistent with other
+  Puppet Labs layouts, such as Puppet Server.
+
+    This means the following paths have changed:
+
+    * Primary configuration: /etc/puppetdb is now /etc/puppetlabs/puppetdb
+    * Logs: /var/log/puppetdb is now /var/log/puppetlabs/puppetdb
+    * Binaries: /usr/bin is now /opt/puppetlabs/bin
+    * Data: /var/lib/puppetdb is now /opt/puppetlabs/server/data/puppetdb
+    * Rundir: /var/run/puppetdb is now /var/run/puppetlabs/puppetdb
+
+    During upgrade we have written a script that will migrate your configuration
+    to the new locations. After upgrade you should double check the migrated
+    configuration to ensure everything is correct. Of course we recommend the
+    use of the [PuppetDB Puppet Module][puppetdb-module] version 5, as it will
+    automate a lot of this kind of management and has been modified specifically
+    for PuppetDB 3 support by default.
+
+* We have retired PostgreSQL earlier than 9.4. We recommended before upgrading
+  Puppet to upgrade PostgreSQL to version 9.4 or newer with the
+  [`pg_trgm`][pg_trgm] extension enabled, as explained
+  [here][configure_postgres], and we have officially retired versions earlier
+  than 9.4. The official PostgreSQL repositories are recommended for packaging, see the
+  ([YUM](https://wiki.postgresql.org/wiki/YUM\_Installation)) instructions or the
+  ([APT](https://wiki.postgresql.org/wiki/Apt)) instructions for more details.
+
+* PuppetDB now only support Puppet 4 or higher at the moment and we are still looking
+  into what packaging requirements are needed to support older Puppet 3 support.
+  For now you must have installed Puppet 4 from Puppet Collections 1
+  before installing PuppetDB 3.
+
+* Before upgrading, make sure that all of your PuppetDB instances are shut down,
+  and only upgrade one at a time.
+
+* Make sure to upgrade your `puppetdb-terminus` package (on the host where your
+  Puppet Master lives), and restart your master service (this is typically called
+  `puppetserver`). It is worth noting that the `puppetdb-terminus` package has
+  now been renamed to `puppetdb-termini`.
+
+* There are some database schema changes (also known as migrations) provided in
+  this release, that may mean you have to wait while these run after upgrading.
+  Before upgrading you should also ensure you have as much free disk space as
+  you have data, as migrations often require temporary duplication of table
+  data.
```

Comment:
`Before upgrading you should ensure you have as much free disk space as you have data. Database schema changes (also known as migrations) in this release often require temporary duplication of table data and may mean you have wait during the upgrade as these migrations run.`